src/declarativewebcontainer.cpp

```diff
@@ -764,25 +709,13 @@ void DeclarativeWebContainer::releasePage(int tabId, bool virtualize)
         m_webPages->release(tabId, virtualize);
         // Successfully destroyed. Emit relevant property changes.
         if (!m_webPage || m_model->count() == 0) {
-            if (m_canGoBack) {
-                m_canGoBack = false;
-                emit canGoBackChanged();
-            }
-
-            if (m_canGoForward) {
-                m_canGoForward = false;
-                emit canGoForwardChanged();
-            }
-
-            emit contentItemChanged();
-            updateUrl("");
-            updateTitle("");
 
```

Comment:
When a page is destroyed... canGoBack, canGoForward, title, url must be empty.

Thus, I think that over here you should do following:
```
emit canGoBackChanged();
emit canGoForwardChanged();
emit urlChanged();
emit titleChanged();
```